packages/tangle/timemanager.go

```diff
@@ -117,33 +133,54 @@ func (t *TimeManager) synced() bool {
 	return clock.Since(t.lastConfirmedMessage.Time) < t.tangle.Options.SyncTimeWindow
 }
 
+// checks whether the synced state needs to be updated and if so,
+// triggers a corresponding event reflecting the new state.
+func (t *TimeManager) updateSyncedState() {
+	t.lastSyncedMutex.Lock()
+	defer t.lastSyncedMutex.Unlock()
+	if newSynced := t.synced(); t.lastSynced != newSynced {
+		t.lastSynced = newSynced
+		// trigger the event inside the lock to assure that the status is still correct
+		t.Events.SyncChanged.Trigger(&SyncChangedEvent{Synced: newSynced})
+	}
+}
+
 // updateTime updates the last confirmed message.
 func (t *TimeManager) updateTime(marker markers.Marker, newLevel int, transition events.ThresholdEventTransition) {
 	if transition != events.ThresholdLevelIncreased {
 		return
 	}
-	// get message ID of marker
+
 	messageID := t.tangle.Booker.MarkersManager.MessageID(&marker)
 
 	t.tangle.Storage.Message(messageID).Consume(func(message *Message) {
 		t.lastConfirmedMutex.Lock()
 		defer t.lastConfirmedMutex.Unlock()
 
-		if t.lastConfirmedMessage.Time.Before(message.IssuingTime()) {
-			t.lastConfirmedMessage = LastConfirmedMessage{
-				MessageID: messageID,
-				Time:      message.IssuingTime(),
-			}
-
-			if newSynced := t.synced(); t.lastSynced != newSynced {
-				t.lastSynced = newSynced
-				// trigger the event inside the lock to assure that the status is still correct
-				t.Events.SyncChanged.Trigger(&SyncChangedEvent{Synced: newSynced})
-			}
+		if t.lastConfirmedMessage.Time.After(message.IssuingTime()) {
+			return
+		}
+
+		t.lastConfirmedMessage = LastConfirmedMessage{
+			MessageID: messageID,
+			Time:      message.IssuingTime(),
 		}
+
+		t.updateSyncedState()
 	})
 }
 
+// the main loop runs the updateSyncedState at least every synced time window interval to keep the synced state updated
+// even if no updateTime is ever called.
+func (t *TimeManager) mainLoop() {
+	timeutil.NewTicker(t.updateSyncedState, func() time.Duration {
```

Comment:
I personally think:
```go
interval := DefaultSyncTimeWindow
if t.tangle.Options.SyncTimeWindow != 0 {
	interval = t.tangle.Options.SyncTimeWindow
}
```
is a bit nicer than the `func` construct, but I am happy with both